File: moderate/overlapping_rectangles.hs
import System.Environment (getArgs)
import Data.List.Split (splitOn)

within         :: [Int] -> Int -> Int -> Int -> Bool
within xs r p q = xs!!p >= xs!!r && xs!!p <= xs!!(r+2) && xs!!q >= xs!!(r+3) && xs!!q <= xs!!(r+1)

overlapping   :: [Int] -> String
overlapping xs | within xs 0 4 5 || within xs 0 6 7 || within xs 0 4 7 || within xs 0 6 5 || within xs 4 0 1 || within xs 4 2 3 || within xs 4 0 3 || within xs 4 2 1 = "True"
               | otherwise = "False"

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (overlapping . map read . splitOn ",") $ lines input

File: moderate/overlapping_rectangles.lua
function within(a, r, p, q)
  return a[p] >= a[r] and a[p] <= a[r+2] and a[q] >= a[r+3] and a[q] <= a[r+1]
end

for line in io.lines(arg[1]) do
  a = {}
  for i in line:gmatch("[-]?%d+") do a[#a + 1] = tonumber(i) end
  print((within(a, 1, 5, 6) or within(a, 1, 7, 8) or within(a, 1, 5, 8) or within(a, 1, 7, 6) or
         within(a, 5, 1, 2) or within(a, 5, 3, 4) or within(a, 5, 1, 4) or within(a, 5, 3, 2)) and
        "True" or "False")
end
